Companion PR of "Add check generated files and check code style test components"

ronald-cron-arm opened the pull request:
## Description
Companion PR of https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/201.
Adapt to the changes in the framework done for https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/201.
Use make_generated_files.py to check make and cmake generated files instead of `check-generated-files.sh`.

## PR checklist
- [x] **changelog** not required because: only internal changes
- [x] **development PR** provided here
- [x] **TF-PSA-Crypto PR** provided Mbed-TLS/tf-psa-crypto#201 
- [x] **framework PR** provided Mbed-TLS/mbedtls-framework#149
- [x] **3.6 PR** provided #10069
- [x] **2.28 PR** not required because: no impact on 2.28 
- **tests**  not required because: only internal changes

valeriosetti commented on tests/scripts/components-basic-checks.sh:
```diff
@@ -17,20 +17,38 @@ component_check_recursion () {
-    tests/scripts/check-generated-files.sh
+    $FRAMEWORK/scripts/make_generated_files.py --check
+
+    cd tf-psa-crypto
```
```suggestion
    cd $TF_PSA_CRYPTO_ROOT_DIR
```
here and also on line 50?

valeriosetti reviewed (changes requested):
LGTM. Only a minor comment

gabor-mezei-arm reviewed (dismissed):
LGTM

ronald-cron-arm commented:
I've rebased on top of development HEAD, updated the two commits "Update framework pointer" and "Update TF-PSA-Crypto pointer" and moved the "Update TF-PSA-Crypto pointer" at the end.

mpg reviewed (approved):
Re-approving based on Gabor's approval of edabc0933a978e2b055ff75ae935045aad7f0264 - only changes are in the crypto pointer, which I've verified points to the head of 201.
